sycl/test-e2e/Adapters/sycl-ls-gpu-default-any.cpp

```diff
@@ -1,9 +1,8 @@
 // REQUIRES: gpu
-// REQUIRES: build-and-run-mode
 
 // TODO: Remove unsetting SYCL_DEVICE_FILTER when feature is dropped
-// RUN: env --unset=SYCL_DEVICE_FILTER  --unset=ONEAPI_DEVICE_SELECTOR sycl-ls --verbose >%t.default.out
-// RUN: FileCheck %s --check-prefixes=CHECK-GPU-BUILTIN,CHECK-GPU-CUSTOM --input-file %t.default.out
+// RUN: %if run-mode %{ env --unset=SYCL_DEVICE_FILTER  --unset=ONEAPI_DEVICE_SELECTOR sycl-ls --verbose >%t.default.out %}
+// RUN: %if run-mode %{ FileCheck %s --check-prefixes=CHECK-GPU-BUILTIN,CHECK-GPU-CUSTOM --input-file %t.default.out %}
```

Comment:
I'd prefer a pipe so that not `%if run-mode` would be necessary: `%{run-unfiltered-devices} sycl-ls | FileCheck`